src/main/java/liquibase/ext/neo4j/lockservice/Neo4jLockService.java

```diff
@@ -0,0 +1,124 @@
+package liquibase.ext.neo4j.lockservice;
+
+import liquibase.Scope;
+import liquibase.database.Database;
+import liquibase.exception.DatabaseException;
+import liquibase.exception.LiquibaseException;
+import liquibase.exception.LockException;
+import liquibase.executor.Executor;
+import liquibase.executor.ExecutorService;
+import liquibase.ext.neo4j.database.Neo4jDatabase;
+import liquibase.lockservice.DatabaseChangeLogLock;
+import liquibase.lockservice.LockService;
+import liquibase.statement.SqlStatement;
+import liquibase.statement.core.RawSqlStatement;
+
+import java.util.Collections;
+import java.util.UUID;
+
+public class Neo4jLockService implements LockService {
+
+    private static final String CONSTRAINT_NAME = "unique_liquibase_lock";
+    // this query will only work with Neo4j 4.x (not 3.x, as db.constraints() does not yield the 'name' column)
+    private static final String CHECK_LOCK_CONSTRAINT_QUERY = String.format("CALL db.constraints() YIELD name\n" +
+            "WITH collect(name) AS names\n" +
+            "RETURN NONE(name IN names WHERE name = \"%1$s\" OR name = \"%1$s_bc\") AS result", CONSTRAINT_NAME);
+
+    private Database database;
+
+    @Override
+    public int getPriority() {
+        return PRIORITY_DATABASE;
+    }
+
+    @Override
+    public boolean supports(Database database) {
+        return database instanceof Neo4jDatabase;
+    }
+
+    @Override
+    public void setDatabase(Database database) {
+        this.database = database;
+    }
+
+    @Override
+    public void setChangeLogLockWaitTime(long changeLogLockWaitTime) {
+
+    }
+
+    @Override
+    public void setChangeLogLockRecheckTime(long changeLogLocRecheckTime) {
+
+    }
+
+    @Override
+    public boolean hasChangeLogLock() {
+        return true;
+    }
+
+    @Override
+    public void waitForLock() throws LockException {
+
+    }
+
+    @Override
+    public void init() throws DatabaseException {
+        try {
+            Executor executor = Scope.getCurrentScope().getSingleton(ExecutorService.class).getExecutor("jdbc",  database);
```

Comment:
@nvoxland please let me know if I should use another API to run a query with results. I could not find a way to do that with `liquibase.database.Database`

Reply:
That is the best route for now. We'll simplify the interface in an upcoming release

Reply:
Duly noted, thanks!